### ISSUE TITLE
Indented heredoc issue when used in a function

### ISSUE THREAD

**Issue by atoomic** [perltidy/perltidy#82]
The current program cannot be perltidy 
Extract from `t/snippets/expect/heredoc_indented_sub.def` 
# note need to remove the v5.14 and test() call, just here for illustration
```perl
use v5.30;
sub test {

    my $fh = \*STDERR;

    say ${fh} <<~'EOS';
    # hello world
    EOS

    say ${fh} <<~"EOS";
    boom  = 66
    EOS

    return;
}

test();
```

```
> perl -cw t/snippets/expect/heredoc_indented_sub.def
t/snippets/expect/heredoc_indented_sub.def syntax OK
```

```
> perl5.30 t/snippets/expect/heredoc_indented_sub.def
# hello world

boom  = 66

```

but when running perltidy on it
```
> perltidy t/snippets/expect/heredoc_indented_sub.def
## Please see file t/snippets/expect/heredoc_indented_sub.def.ERR
```

```
> cat t/snippets/expect/heredoc_indented_sub.def.ERR
Perltidy version is 20211029

t/snippets/expect/heredoc_indented_sub.def: Begin Error Output Stream

12: EOS
    ^
found bareword where operator expected

Missing ';' or ',' above?
```

**Comment by atoomic** [comment]
we can close this the issue was using `say ${fh}` instead of ` say {$fh}`

**Comment by shancock9** [comment]
The problem was that perltidy isn't correctly treating something like ${fh} as a possible indirect object. I've written a simple update which fixes this, but since this is low priority I'll wait until after the next release to install and test it.

**Comment by atoomic** [comment]
thanks

**Comment by shancock9** [comment]
This is fixed in version 20220217. Thanks again for reporting it.
